packages/runtime/plugin/src/withIosAppDelegateResetKeychain.ts

```diff
@@ -0,0 +1,91 @@
+// Inspired by https://github.com/expo/expo/blob/03e99016c9c5b9ad47864b204511ded2dec80375/packages/%40expo/config-plugins/src/ios/Maps.ts#L6
+import { ConfigPlugin, withAppDelegate } from '@expo/config-plugins'
+import { mergeContents, MergeResults } from '@expo/config-plugins/build/utils/generateCode'
+
+const RESET_KEYCHAIN_FUNCTION = `
+// Use same key as react-native-secure-key-store
+// so we don't reset already working installs
+static NSString * const kHasRunBeforeKey = @"RnSksIsAppInstalled";
+
+// Reset keychain on first app run, this is so we don't run with leftover items
+// after reinstalling the app
+static void resetKeychainIfNecessary()
+{
+  NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+  if ([defaults boolForKey:kHasRunBeforeKey]) {
+    return;
+  }
+  
+  NSArray *secItemClasses = @[(__bridge id)kSecClassGenericPassword,
+                              (__bridge id)kSecAttrGeneric,
+                              (__bridge id)kSecAttrAccount,
+                              (__bridge id)kSecClassKey,
+                              (__bridge id)kSecAttrService];
+  for (id secItemClass in secItemClasses) {
+    NSDictionary *spec = @{(__bridge id)kSecClass:secItemClass};
+    SecItemDelete((__bridge CFDictionaryRef)spec);
+  }
+  
+  [defaults setBool:YES forKey:kHasRunBeforeKey];
+  [defaults synchronize];
+}
+`
+
+const METHOD_INVOCATION_BLOCK = `resetKeychainIfNecessary();`
+
+// https://regex101.com/r/nHrTa9/1/
+const INVOCATION_LINE_MATCHER =
+  /-\s*\(BOOL\)\s*application:\s*\(UIApplication\s*\*\s*\)\s*\w+\s+didFinishLaunchingWithOptions:/g
+
+function addResetKeychainFunction(src: string): MergeResults {
+  const newSrc = ['#if canImport(GoogleMaps)', 'import GoogleMaps', '#endif']
```

Comment:
checking if we need to mention google maps here?

Reply:
Good catch, forgot to remove that 😅